src/lib/transition_router/transition_router.ml

```diff
@@ -542,9 +542,17 @@ let run ?(sync_local_state = true) ?(cache_exceptions = false)
   let verified_transition_reader, verified_transition_writer =
     let name = "verified transitions" in
     create_buffered_pipe ~name
-      ~f:(fun ( `Transition (head : Mina_block.Validated.t)
-              , _
-              , `Valid_cb valid_cb ) ->
+      ~f:(fun (message, _, `Valid_cb valid_cb) ->
+        let head =
+          match message with
+          | `Block b ->
+              b
+          | `Transition t ->
+              t
+          | `Header _ ->
+              (* TODO this may introduce a DOS add proper error handling*)
+              failwith "Header not supported"
```

Comment:
I wanted some guidance on what to do for this. I understand that running failwith as a result of messages over the network is DOS 101.